contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -66,16 +86,41 @@ contract TrueCreditAgency is UpgradeableClaimable {
         emit PoolAllowed(pool, isAllowed);
     }
 
-    function creditScoreAdjustmentRate(address borrower) public view returns (uint256) {
-        uint8 score = creditOracle.getScore(borrower);
+    function updateCreditScore(ITrueFiPool2 pool, address borrower) external {
+        uint8 oldScore = creditScore[pool][borrower];
+        uint8 newScore = creditOracle.getScore(borrower);
+        if (oldScore == newScore) {
+            return;
+        }
+        require(newScore > 0, "TrueCreditAgency: Cannot set credit score to 0");
```

Comment:
Why?

Reply:
Division by 0 protection, we shouldn't give score 0 to any existing borrower

Reply:
What if we dewhitelist borrower, he's removed from borrowers list and his score goes to zero? IMO sounds like something that might happen.

Reply:
We should just set it to 1

Reply:
Maybe we can use saturating operators that return max uint256 interest for borrowers with credit score 0. It will be cut off by the 500% interest cap anyway.

Protection from divide by zero errors should happen as close as possible to the source, to limit the spread of extraneous error-checking. I think there's no need to separately handle borrowers with 0 credit score, as far as that's feasible.